firedrake/interpolation.py

```diff
@@ -33,12 +33,65 @@
 __all__ = (
     "interpolate",
     "Interpolator",
+    "Interpolate",
     "DofNotDefinedError",
     "CrossMeshInterpolator",
     "SameMeshInterpolator",
 )
 
 
+class Interpolate(ufl.Interpolate):
+
+    def __init__(self, expr, v, interp_data=None):
```

Comment:
Is there a good reason why `interp_data` is not `**kwargs` given it's an optional dictionary?

Reply:
Yes, that's clearer and cleaner. It distinguishes between the mathematical objects involved in the interpolation and the interpolation data specific to the interpolation computation. That's similar to what we do with solver parameters in `solve`. For example, one would write:

```
I = Interpolate(expr, V, interp_data={"subset": ...,
                                      "default_missing_val": ...,
                                      "allow_missing_dofs": ...})
```

instead of 

```
I = Interpolate(expr, V, subset=..., default_missing_val=..., allow_missing_dofs=...)
```

Reply:
Makes sense. Maybe it should be called something like `compute_kwargs` then?